lodash.js

```diff
@@ -5868,6 +5870,42 @@
     });
 
     /**
+     * Gets the next value on a wrapped object in ES2015 iterator
+     * format (<link>). Useful for `for..of` loops and using with
+     * some ES2015 features.
+     *
+     * @name next
+     * @memberOf _
+     * @category Chain
+     * @returns {*} Returns the next value.
+     * @example
+     *
+     * var array = [1, 2];
+     * var wrapped = _(array);
+     *
+     * console.log(wrapped.next());
+     * // => {done: false, value: 1}
+     *
+     * console.log(wrapped.next());
+     * // => {done: false, value: 2}
+     *
+     * console.log(wrapped.next());
+     * // => {done: true, value: undefined}
+     *
+     * var array = [1, 2];
+     * var wrapped = _(array);
+     *
+     * // Use with an ES6 methods which accept an iterable
+     * console.log(Array.from(wrapped))
+     * // => [1, 2]
+     */
+    function wrapperNext() {
+      if (this.__value__ == null) this.__value__ = this.value();
```

Comment:
Sure,
1) The `__value__` can be set by `.value()` and returned by `.value()` when it exists.
2) Since the resulting value, `__value__`, is known we can delete `__wrapped__` as all  computations have completed (to this point).